test/steps/kogitoexplainability.go

```diff
@@ -0,0 +1,130 @@
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package steps
+
+import (
+	"encoding/json"
+	"fmt"
+	"net/http"
+
+	"github.com/cucumber/godog"
+	"github.com/kiegroup/kogito-cloud-operator/test/framework"
+	"github.com/kiegroup/kogito-cloud-operator/test/steps/mappers"
+	bddtypes "github.com/kiegroup/kogito-cloud-operator/test/types"
+)
+
+/*
+	DataTable for Explainability:
+	| runtime-request | cpu/memory  | value                     |
+	| runtime-limit   | cpu/memory  | value                     |
+	| runtime-env     | varName     | varValue                  |
+	| kafka           | externalURI | kafka-bootstrap:9092      |
+	| kafka           | instance    | external-kafka            |
+*/
+
+func registerKogitoExplainabilityServiceSteps(ctx *godog.ScenarioContext, data *Data) {
+	ctx.Step(`^Install Kogito Explainability with (\d+) replicas$`, data.installKogitoExplainabilityServiceWithReplicas)
+	ctx.Step(`^Install Kogito Explainability with (\d+) replicas with configuration:$`, data.installKogitoExplainabilityServiceWithReplicasWithConfiguration)
+	ctx.Step(`^Kogito Explainability has (\d+) pods running within (\d+) minutes$`, data.kogitoExplainabilityHasPodsRunningWithinMinutes)
+	ctx.Step(`^Explainability result is available in the Trusty service within (\d+) minutes$`, data.explainabilityResultIsAvailable)
+}
+
+func (data *Data) installKogitoExplainabilityServiceWithReplicas(replicas int) error {
+	explainability := framework.GetKogitoExplainabilityResourceStub(data.Namespace, replicas)
+	return framework.InstallKogitoExplainabilityService(data.Namespace, framework.GetDefaultInstallerType(), &bddtypes.KogitoServiceHolder{KogitoService: explainability})
+}
+
+func (data *Data) installKogitoExplainabilityServiceWithReplicasWithConfiguration(replicas int, table *godog.Table) error {
+	explainability := &bddtypes.KogitoServiceHolder{
+		KogitoService: framework.GetKogitoExplainabilityResourceStub(data.Namespace, replicas),
+	}
+
+	if err := mappers.MapKogitoServiceTable(table, explainability); err != nil {
+		return err
+	}
+
+	return framework.InstallKogitoExplainabilityService(data.Namespace, framework.GetDefaultInstallerType(), explainability)
+}
+
+func (data *Data) kogitoExplainabilityHasPodsRunningWithinMinutes(podNb, timeoutInMin int) error {
+	return framework.WaitForKogitoExplainabilityService(data.Namespace, podNb, timeoutInMin)
+}
+
+type executionsResponse struct {
+	Executions []execution `json:"executions"`
+}
+
+type execution struct {
+	ExecutionID string `json:"executionId"`
+}
+
+func (data *Data) explainabilityResultIsAvailable(timeoutInMin int) error {
```

Comment:
This test is not supposed to work at the moment, since the implementations in the trusty and explainability services are still in progress
https://github.com/kiegroup/kogito-runtimes/pull/720
https://github.com/kiegroup/kogito-apps/pull/410